WooCommerce/Classes/Styles/PaddedLabel.swift

```diff
@@ -16,11 +16,14 @@ class PaddedLabel: UILabel {
         super.drawText(in: rect.inset(by: textInsets))
     }
 
-    override var intrinsicContentSize: CGSize {
-        var intrinsicSuperViewContentSize = super.intrinsicContentSize
-        intrinsicSuperViewContentSize.height += textInsets.top + textInsets.bottom
-        intrinsicSuperViewContentSize.width += textInsets.left + textInsets.right
-        return intrinsicSuperViewContentSize
+    override func textRect(forBounds bounds: CGRect, limitedToNumberOfLines numberOfLines: Int) -> CGRect {
```

Comment:
This is a genuine question: my knowledge of the intricacies of the autolayout system is very limited, but my understanding was that when the system relies heavily on intrinsicContentSize. Would it make sense to actually implement both intrinsicContentSize and textRect:forBounds?

@mindgraffiti You know much more than me about this, what do you think?

Reply:
After I first got the full text to show with `textRect:forBounds`, I kept the pre-existing `intrinsicContentSize` override. However, during my testing I noticed there's extra padding both vertically and horizontally.

overriding `textRect:forBounds` and `intrinsicContentSize` | overriding `textRect:forBounds` but not `intrinsicContentSize`
--- | ---
![Simulator Screen Shot - iPhone SE - 2019-06-21 at 15 32 44](https://user-images.githubusercontent.com/1945542/59948038-ef955280-943c-11e9-9eb2-45a38289304f.png) | ![Simulator Screen Shot - iPhone SE - 2019-06-21 at 15 26 31](https://user-images.githubusercontent.com/1945542/59948042-f1f7ac80-943c-11e9-84d3-1016a730a15d.png)

My guess is that both overrides apply the label padding from `textInsets`, and `textRect:forBounds` specifies where the text is drawn. Though I'm not an expert about these methods and would like to learn how they work differently in more depth

Reply:
So I read the documentation on `textRect:forBounds` and `intrinsicContentSize`, and `UILabel` as well. There were no answers to be found there. I'm still researching this (I still need to read the Draw lifecycle of a UIView), but I have doubts that this is the solution we want to take.

I did read the SO solution, and this implementation is also missing `didSet { invalidateIntrinsicContentSize() }` mentioned [here](https://stackoverflow.com/a/33573850/4150507).

Reply:
I also found the solution in this PR pretty hacky, I don't feel so confident when I cannot explain why `intrinsicContentSize` doesn't just work (most SO answers suggested just overriding `intrinsicContentSize` as before this PR). Another alternative is to not try adding padding to `UILabel` (since it is not supported natively): just have a `UIView` that contains a `UILabel` subview with insets. I was thinking about using Auto Layout, then checked the Slack discussion where Nate suggested the [WPiOS implementation](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Views/PaddedLabel.swift). I'll try with Auto Layout first, and then the WPiOS implementation

> I did read the SO solution, and this implementation is also missing didSet { invalidateIntrinsicContentSize() } mentioned here. Was there something you noticed that made you decide not to include it?

great catch! I was trying to make minimal changes since I was uncertain about the changes, but yes I believe `didSet { invalidateIntrinsicContentSize() }` is necessary if going with the `textRect:forBounds:` approach

Reply:
>Eric suggested the WPiOS implementation. 

>I'll try with Auto Layout first, and then the WPiOS implementation

Thank you! That sounds like a better alternative.